apps/data-brittler/src/jobs/subscriptions/benefits.rs

```diff
@@ -0,0 +1,2168 @@
+//         {
+//             "id": "62f97db2e46eb00e438a696b",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 3 Month": [
+//         {
+//             "id": "62f97e19e46eb00e438a696c",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 6 Month": [
+//         {
+//             "id": "62f97e71e46eb00e438a696d",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 9 Month": [
+//         {
+//             "id": "62f97ebee46eb00e438a696e",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 1 Year": [
+//         {
+//             "id": "62f97f19e46eb00e438a696f",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 1.25 Year": [
+//         {
+//             "id": "637d53962863630a2d06e27a",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 1.5 Year": [
+//         {
+//             "id": "637d54582863630a2d06e27b",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 2 Year": [
+//         {
+//             "id": "64dfeec02040c6754787d929",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 2.25 Year": [
+//         {
+//             "id": "64dff4652040c6754787d92a",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 2.5 Year": [
+//         {
+//             "id": "64dff95e2040c6754787d92f",
+//             "kind": "badge"
+//         }
+//     ],
+//     "Subscriber 2.75 Year": [
+//         {
+//             "id": "64dffd6b2040c6754787d930",
+//             "kind": "badge"
+//         }
+//     ]
+// }
+
+use shared::database::duration::DurationUnit;
+use shared::database::entitlement::{EntitlementEdgeKind, EntitlementEdgeManagedBy};
+use shared::database::product::codes::RedeemCodeId;
+use shared::database::product::special_event::{SpecialEvent, SpecialEventId};
+use shared::database::product::{SubscriptionBenefit, SubscriptionBenefitCondition, SubscriptionBenefitId, TimePeriod};
+use shared::database::role::RoleId;
+use shared::database::user::UserId;
+
+pub struct BenefitWithEntitlements {
+	pub benefit: SubscriptionBenefit,
+	pub entitlements: Vec<EntitlementEdgeKind>,
+}
+
+impl BenefitWithEntitlements {
+	pub fn new(name: &str, condition: SubscriptionBenefitCondition, entitlements: Vec<EntitlementEdgeKind>) -> Self {
+		Self {
+			benefit: SubscriptionBenefit {
+				id: SubscriptionBenefitId::new(),
+				name: name.to_string(),
+				condition,
+			},
+			entitlements,
+		}
+	}
+}
+
+pub fn sub_badges_benefits() -> impl IntoIterator<Item = BenefitWithEntitlements> {
+	[
+		BenefitWithEntitlements::new(
+			"Subscriber Pack",
+			SubscriptionBenefitCondition::Duration(DurationUnit::Months(1)),
+			vec![
+				EntitlementEdgeKind::Badge {
+					badge_id: "62f97c05e46eb00e438a696a".parse().unwrap(),
+				},
+				EntitlementEdgeKind::Role {
+					role_id: "6076a86b09a4c63a38ebe801".parse().unwrap(),
+				},
```

Comment:
You only get the role after one month?

Reply:
In the calculation we will ceil

Reply:
Wouldn't that mean you get the one month badge without being subbed for one month?